Bluetooth: host: free the adv instance if start legacy advertise fail

anchao opened the pull request:
Bluetooth: host: free the adv instance if start legacy advertise fail

the adv instance should be delete correctly if advertise fail,
Otherwise the instance will unable to allocate if start the legacy again

Signed-off-by: chao an <anchao@xiaomi.com>

joerchan commented on subsys/bluetooth/host/hci_core.c:
```diff
@@ -7778,6 +7779,11 @@ int bt_le_adv_start_legacy(const struct bt_le_adv_param *param,
 	return 0;
+
+failed:
+	adv_delete_legacy();
```
Thank you for the fix, although I would suggest to do it a bit different.
In `bt_le_adv_start`:
```
	err = bt_le_adv_start_legacy(param, ad, ad_len, sd, sd_len);
	if (err) {
		adv_delete_legacy();
	}

	return err;
```

That is similar to how it is handled for extended advertise case.

anchao replied:
Sorry for reply late, I moved the adv instance allocate out of the le_adv_start_legacy(), so as to reuse and unify the adv_new_legacy() and adv_delete_legacy()

anchao committed:
Bluetooth: host: free the adv instance if start legacy advertise fail

the adv instance should be delete correctly if advertise fail,
Otherwise the instance will unable to allocate if start the legacy again

Signed-off-by: chao an <anchao@xiaomi.com>

joerchan commented on subsys/bluetooth/host/hci_core.c:
```diff
@@ -8129,24 +8128,25 @@ int bt_le_adv_start(const struct bt_le_adv_param *param,
+	}
 
+	if (IS_ENABLED(CONFIG_BT_EXT_ADV) &&
+			BT_FEAT_LE_EXT_ADV(bt_dev.le.features)) {
```
Please fix the indentation here, should be 1 tab + 4 spaces, not 3 tabs.

anchao replied:
Got

joerchan reviewed (changes requested):
Looks good, just need an indentation fix.
